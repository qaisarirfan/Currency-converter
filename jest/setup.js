jest.mock('react-native-reanimated', () => {
  const Reanimated = require('react-native-reanimated/mock');

  // The mock for `call` immediately calls the callback which is incorrect
  // So we override it with a no-op
  Reanimated.default.call = () => {};

  return Reanimated;
});

jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useFocusEffect: () => jest.fn(),
    useRoute: () => jest.fn(),
    useNavigation: () => ({
      navigation: () => jest.fn(),
      canGoBack: () => jest.fn(),
      goBack: () => jest.fn(),
      push: () => jest.fn(),
    }),
  };
});

jest.mock('react-native-bootsplash', () => ({
  hide: jest.fn(),
  show: jest.fn(),
}));

jest.mock('react-native-localize', () => ({
  getLocales: () => [
    { countryCode: 'GB', languageTag: 'en-GB', languageCode: 'en', isRTL: false },
    { countryCode: 'US', languageTag: 'en-US', languageCode: 'en', isRTL: false },
    { countryCode: 'FR', languageTag: 'fr-FR', languageCode: 'fr', isRTL: false },
  ],

  getNumberFormatSettings: () => ({
    decimalSeparator: '.',
    groupingSeparator: ',',
  }),

  getCalendar: () => 'gregorian', // or "japanese", "buddhist"
  getCountry: () => 'US', // the country code you want
  getCurrencies: () => ['USD', 'EUR'], // can be empty array
  getTemperatureUnit: () => 'celsius', // or "fahrenheit"
  getTimeZone: () => 'Europe/Paris', // the timezone you want
  uses24HourClock: () => true,
  usesMetricSystem: () => true,

  addEventListener: jest.fn(),
  removeEventListener: jest.fn(),
}));

jest.mock('react-native-keyboard-aware-scroll-view', () => {
  const KeyboardAwareScrollView = require('react-native').ScrollView;
  return { KeyboardAwareScrollView };
});

jest.mock('react-native-gesture-handler', () => {
  // eslint-disable-next-line global-require
  const View = require('react-native/Libraries/Components/View/View');
  return {
    Swipeable: View,
    DrawerLayout: View,
    State: {},
    ScrollView: View,
    Slider: View,
    Switch: View,
    TextInput: View,
    ToolbarAndroid: View,
    ViewPagerAndroid: View,
    DrawerLayoutAndroid: View,
    WebView: View,
    NativeViewGestureHandler: View,
    TapGestureHandler: View,
    FlingGestureHandler: View,
    ForceTouchGestureHandler: View,
    LongPressGestureHandler: View,
    PanGestureHandler: View,
    PinchGestureHandler: View,
    RotationGestureHandler: View,
    /* Buttons */
    RawButton: View,
    BaseButton: View,
    RectButton: View,
    BorderlessButton: View,
    /* Other */
    FlatList: View,
    gestureHandlerRootHOC: jest.fn(),
    Directions: {},
  };
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
