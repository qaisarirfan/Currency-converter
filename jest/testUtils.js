import React from "react"
import {render} from "react-native-testing-library"
import {Provider} from "react-redux"
import configureStore from "redux-mock-store"
import ThemeContextProvider from "../src/ContextUtils/ThemeContext"
import ConversionContextProvider from "../src/ContextUtils/ConversionContext"
import {initialState as themesState} from "../src/redux/reducers/themes"

const INITIAL_STATE = {
  themes: themesState,
}
const mockStore = configureStore([])
const store = mockStore(INITIAL_STATE)

const reduxRender = (ui, options) =>
  render(ui, {
    wrapper: ({children}) => (
      <Provider store={store}>
        <ThemeContextProvider>
          <ConversionContextProvider>{children}</ConversionContextProvider>
        </ThemeContextProvider>
      </Provider>
    ),
    ...options,
  })

// re-export everything for convenience
export * from "react-native-testing-library"

// override render method
export {reduxRender as render, store}
