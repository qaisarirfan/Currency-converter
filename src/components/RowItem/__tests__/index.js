import React from "react"
import {render, cleanup, store} from '../../../../jest/testUtils';
import {RowItem, RowSeparator} from ".."

afterEach(() => {
  cleanup();
  store.clearActions();
});

describe("<RowItem />", () => {
  const createProps = () => ({
    title: "test",
  })
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<RowItem {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })

})

describe("<RowSeparator />", () => {
  const createProps = () => ({
    style: {
      backgroundColor: "#fff"
    },
  })
  it("Does not explode", () => {
    const props = createProps()
    const component = render(<RowSeparator {...props} />).toJSON()
    expect(component).toMatchSnapshot()
  })
})
