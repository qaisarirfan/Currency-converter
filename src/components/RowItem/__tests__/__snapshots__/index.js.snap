// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RowItem /> Does not explode 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#8499a2",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingHorizontal": 20,
      "paddingVertical": 16,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#eaedef",
        "fontSize": 16,
      }
    }
  >
    test
  </Text>
</View>
`;

exports[`<RowSeparator /> Does not explode 1`] = `
<View
  style={
    Object {
      "height": 0.5,
    }
  }
/>
`;
